extensions/amp-consent/0.1/tcf-api-command-manager.js

```diff
@@ -60,13 +75,80 @@ export class TcfApiCommandManager {
         this.handlePingEvent_(payload, win);
         break;
       case TCF_POST_MESSAGE_API_COMMANDS.GET_TC_DATA:
+        this.handleGetTcData_(payload, win);
+        break;
       case TCF_POST_MESSAGE_API_COMMANDS.ADD_EVENT_LISTENER:
       case TCF_POST_MESSAGE_API_COMMANDS.REMOVE_EVENT_LISTENER:
       default:
         return;
     }
   }
 
+  /**
+   * @return {!Promise<Array>}
+   */
+  getTcDataPromises_() {
+    const consentStringInfoPromise = this.policyManager_.getConsentStringInfo(
+      'default'
+    );
+    const metadataPromise = this.policyManager_.getConsentMetadataInfo(
+      'default'
+    );
+    const sharedDataPromise = this.policyManager_.getMergedSharedData(
+      'default'
+    );
+
+    return Promise.all([
+      metadataPromise,
+      sharedDataPromise,
+      consentStringInfoPromise,
+    ]);
+  }
+
+  /**
+   * Create minimal PingReturn object. Send to original iframe
+   * once object has been filled.
+   * @param {!Object} payload
+   * @param {!Window} win
+   */
+  handleGetTcData_(payload, win) {
+    this.getTcDataPromises_().then((arr) => {
+      const returnValue = this.getMinimalTcData_(arr[0], arr[1], arr[2]);
```

Comment:
nit: can you use array destructuring to name what arr[0] thru arr[2] are? e.g. `const [metadata, sharedData, tcString] = arr;`

Reply:
Array restructuring is not allowed in classic AMP :(

Would it be be better to have `getTcDataPromises_()` return an object or a Def and then destructure that?

Reply:
Weird that we support object destructuring, but not array destructuring. Leave it as is then.